src/pages/MyProfile.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  Alert,
+  Button,
+  Col,
+  Container,
+  ListGroup,
+  Row,
+} from 'react-bootstrap';
+import { toggleReservationState } from '../redux/rockets/rocketsSlice';
+
+function MyProfile() {
+  const rockets = useSelector((state) => state.rockets);
+  const missions = useSelector((state) => state.missions.missions);
+  const dispatch = useDispatch();
+
+  let joinedMissions = [];
+  if (missions) {
+    joinedMissions = missions.length ? missions.filter((mission) => mission.status) : [];
+  }
+
+  const cancelRocketReservation = (id) => {
+    dispatch(toggleReservationState(id));
+  };
+
+  const filterReservedRockets = (rocks) => {
+    const reserved = rocks.filter((rocket) => rocket.isReserved);
+
+    return reserved.length > 0
+      ? reserved.map((rocket) => (
+        <ListGroup.Item key={rocket.id}>
+          <div className="d-flex justify-content-between align-items-center">
+            {rocket.rocketName}
+            <div>
+              <a className="btn btn-info me-3" href={rocket.wikipedia} target="_blank" rel="noreferrer">Read more</a>
+              <Button variant="danger" onClick={() => cancelRocketReservation(rocket.id)}>Cancel Reservation</Button>
+            </div>
+
+          </div>
+        </ListGroup.Item>
+      ))
+      : (<Alert variant="info">No rockets reserved</Alert>);
+  };
+
+  return (
+    <Container className="mt-5">
+      <Row>
+        <Col>
+          <h1>My Missions</h1>
+          <ListGroup>
+            {joinedMissions.length ? joinedMissions.map((mission) => (
+              <ListGroup.Item key={mission.mission_id}>{mission.mission_name}</ListGroup.Item>
+            )) : <h6 className="text-danger">No Joined Group</h6>}
+          </ListGroup>
```

Comment:
## NICE JOB 👏 YOUR PROJECT IS WELL STRUCTURED, WELL DESIGNED, WELL TESTED, WELL... I have added my start ⭐  on the repo 😄 . sure you deserve it 👏 , team.

## [OPTIONAL]
- [ ] I would really suggest you implement the leave mission in your profile page just like you did with the `cancel reservation` on the profile page.
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/65068771/216525069-a4e1a285-7496-4a1f-a830-c0081cfa36e7.png">